tempodb/backend/block_meta.go

```diff
@@ -63,15 +65,16 @@ type DedicatedColumn struct {
 	Type string `yaml:"type" json:"type"`
 }
 
-func NewBlockMeta(tenantID string, blockID uuid.UUID, version string, encoding Encoding, dataEncoding string) *BlockMeta {
+func NewBlockMeta(tenantID string, blockID uuid.UUID, version string, encoding Encoding, dataEncoding string, dedicatedColumns ...DedicatedColumn) *BlockMeta {
```

Comment:
I'd prefer to use `dedicatedColumns []DedicatedColumn` instead of a variadic parameter here
* It makes it easier to change or extend the function signature
* In this situation I find it actually positive too explicitly pass in `nil` or empty slice when no dedicated columns are used

Reply:
Changed. Added a new method when necessary, and changed variadic params to slices.